ipfs-cluster-service/lock.go

```diff
@@ -0,0 +1,69 @@
+// lock logic heavily inspired by go-ipfs/repo/fsrepo/lock/lock.go
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strings"
+	"syscall"
+
+	filelock "go4.org/lock"
+)
+
+// The name of the file used for locking
+const lockFileName = "cluster.lock"
+
+// Global file, close to relinquish lock access
+var lockCloser io.Closer
```

Comment:
We should avoid global variables. Let's wrap the locking implementation in a `LockHelper` or `Locker` object which can lock/unlock.